### PR TITLE
Improve DestinationRule Security Best Practices

### PR DESCRIPTION
**Pull request by Kmoneal**
* Add instructions for improving security using subjectAltNames which is
not checked by default.
* Add instructions to turn on VERIFY_CERTIFICATE_AT_CLIENT to decrease
friction of checking certificates against a CA.
* Escalate certificate validation that is not being done to a warning to
increase visibility.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by Kmoneal** [comment]
`VERIFY_CERTIFICATE_AT_CLIENT` being reviewed for 1.11: https://github.com/istio/istio/pull/35188
`VERIFY_CERTIFICATE_AT_CLIENT` in master: https://github.com/istio/istio/pull/33472

File: content/en/docs/ops/best-practices/security/index.md [hunk 1]
```diff
@@ -387,8 +387,14 @@ This ensures that even if a client accidentally or maliciously bypasses their si
 Istio offers the ability to [originate TLS](/docs/tasks/traffic-management/egress/egress-tls-origination/) from a sidecar proxy or gateway.
 This enables applications that send plaintext HTTP traffic to be transparently "upgraded" to HTTPS.
 
-Care must be taken when configuring the `DestinationRule`'s `tls` setting to specify the `caCertificates` field.
-When this is not set, the servers certificate will not be verified.
+Care must be taken when configuring the `DestinationRule`'s `tls` setting to specify the `caCertificates` and `subjectAltNames` fields.
+The `caCertificate` can be automatically set as the OS CA certificate by enabling the environment variable `VERIFY_CERTIFICATE_AT_CLIENT=true` on Istiod.
```
Comment (rvennam) [inline content/en/docs/ops/best-practices/security/index.md:391]: OS?

File: content/en/docs/ops/best-practices/security/index.md [hunk 2]
```diff
@@ -387,8 +387,14 @@ This ensures that even if a client accidentally or maliciously bypasses their si
...
+Specifying the `caCertificates` in a `DestinationRule` will take priority and the OS CA Cert will not be used.
```
Comment (howardjohn) [inline content/en/docs/ops/best-practices/security/index.md:392]: should mention you can explicitly set it to "system" on a DR

File: content/en/docs/ops/best-practices/security/index.md (hunk 1 above)
Comment (howardjohn) [reply content/en/docs/ops/best-practices/security/index.md:391]: I would call it "the system's certificate store". Quoted from some Mozilla page

Comment (Kmoneal) [reply content/en/docs/ops/best-practices/security/index.md:391]: I will use `Operating System` instead (if it's a different concern please correct me).

File: content/en/docs/ops/best-practices/security/index.md (hunk 2 above)
Comment (Kmoneal) [reply content/en/docs/ops/best-practices/security/index.md:392]: I'll add that!

File: content/en/docs/ops/best-practices/security/index.md [hunk 3]
```diff
@@ -387,8 +387,14 @@ This ensures that even if a client accidentally or maliciously bypasses their si
...
+
+{{< warning >}}
+When `caCertificates` is not set, the servers certificate will not be verified.
```
Comment (jacob-delgado) [inline content/en/docs/ops/best-practices/security/index.md:395]:
```suggestion
When `caCertificates` is not set, the servers certificate will not be verified against any CA certificate.
```

File: content/en/docs/ops/best-practices/security/index.md [hunk 4]
```diff
@@ -387,8 +387,14 @@ This ensures that even if a client accidentally or maliciously bypasses their si
...
+When `subjectAltNames` is not set, the Subject Alternative Names of the servers certificate will not be verified.
```
Comment (jacob-delgado) [inline content/en/docs/ops/best-practices/security/index.md:396]:
```suggestion
In order to verify the server's certificate it is important that both `caCertificates` and `subjectAltNames` be set.

Verification of the certificate presented by the server against a CA is not sufficient, as the Subject Alternative Names must also be validated.
```

Comment (jacob-delgado) [reply content/en/docs/ops/best-practices/security/index.md:396]: We should also mention that `sni` should be configured as part of a DestinationRule.

**Comment by rvennam** [comment]
@Kmoneal can you resolve the pending comments?

**Review by rvennam — APPROVED** [review]

**Pull request merged**